plugins/auth/index.js

```diff
@@ -39,22 +41,28 @@ exports.register = (server, options, next) => {
         admins: joi.array().default([]),
         scm: joi.object().required()
     }), 'Invalid config for plugin-auth');
+    const scmContexts = server.root.app.userFactory.scm.getScmContexts();
 
     /**
      * Generates a profile for storage in cookie and jwt
      * @method generateProfile
-     * @param  {String}        username Username of the person
-     * @param  {Array}         scope    Scope for this profile (usually build or user)
-     * @param  {Object}        metadata Additonal information to tag along with the login
-     * @return {Object}                 The profile to be stored in jwt and/or cookie
+     * @param  {String}        username   Username of the person
+     * @param  {String}        scmContext Scm to which the person logged in belongs
+     * @param  {Array}         scope      Scope for this profile (usually build or user)
+     * @param  {Object}        metadata   Additonal information to tag along with the login
+     * @return {Object}                   The profile to be stored in jwt and/or cookie
      */
-    server.expose('generateProfile', (username, scope, metadata) => {
+    server.expose('generateProfile', (username, scmContext, scope, metadata) => {
         const profile = Object.assign({
-            username, scope
+            username, scmContext, scope
         }, metadata || {});
+        const scm = server.root.app.userFactory.scm;
+        const scmDisplayName = scm.getDisplayName({ scmContext });
+        const userDisplayName = scmDisplayName ? `${scmDisplayName}:${username}` : username;
 
         // Check admin
-        if (pluginOptions.admins.length > 0 && pluginOptions.admins.includes(username)) {
+        if (pluginOptions.admins.length > 0
+                && pluginOptions.admins.includes(userDisplayName)) {
```

Comment:
That's right. We need not to think the case that `displayName` setting has not set.
I'll fix them.